nilmtk/electric.py

```diff
@@ -294,6 +294,7 @@ def proportion_of_energy(self, other, **loader_kwargs):
         other_ac_types = other_total_energy.keys()
         self_ac_types = total_energy.keys()
         shared_ac_types = set(other_ac_types).intersection(self_ac_types)
+        shared_ac_types = list(shared_ac_types)
```

Comment:
You overwrite the variable in the previous line. Is there a reason for that?

Reply:
@levaphenyl ,Yes we can do this also , also the line shared_ac_types = list(shared_ac_types) wont change anything , just making it to list

Reply:
Please let me know if it requires to be modified

Reply:
And why do you want it to be a list rather than a set? Set is an iterable, already.

Reply:
By pandas, if we take pandas dataframe and try to do indexing with set(data structure) it through error.
This is the error: 'TypeError: Passing a set as an indexer is not supported. Use a list instead.'

Reply:
OK - then, it relates to newer versions of pandas. Please commit in the other PR, using `list` around `set` on the same line.

Reply:
ok , let me raise new one